From ee6bda44d62a5872ad96c7a9b7c16018525875a8 Mon Sep 17 00:00:00 2001
From: pengy <quic_pengy@quicinc.com>
Date: Thu, 23 May 2024 09:52:30 +0800
Subject: [PATCH] lttng-modules: Fix lttng-modules compile issue on android
 kernel5.15

Remove __kernel_old_time_t dup defination.
Fix vm_scan header files include logic.
Add MODULE_IMPORT_NS for kallsyms module.

Upstream-Status: Inappropriate [other]

Change-Id: If220d7821cd41c3be99902f6bd09201b0aaf52a2
Signed-off-by: Lei wang <quic_leiwan@quicinc.com>
---
 include/instrumentation/events/mm_vmscan.h | 3 +++
 src/lttng-syscalls.h                       | 2 +-
 src/wrapper/kallsyms.c                     | 1 +
 3 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/include/instrumentation/events/mm_vmscan.h b/include/instrumentation/events/mm_vmscan.h
index ea6f4b7..34fef6e 100644
--- a/include/instrumentation/events/mm_vmscan.h
+++ b/include/instrumentation/events/mm_vmscan.h
@@ -851,4 +851,7 @@ LTTNG_TRACEPOINT_EVENT_MAP(update_swap_token_priority,
 #endif /* LTTNG_TRACE_MM_VMSCAN_H */
 
 /* This part must be outside protection */
+#undef TRACE_INCLUDE_PATH
+#define TRACE_INCLUDE_PATH instrumentation/events
+#define TRACE_INCLUDE_FILE mm_vmscan
 #include <lttng/define_trace.h>
diff --git a/src/lttng-syscalls.h b/src/lttng-syscalls.h
index a464e69..7d9a130 100644
--- a/src/lttng-syscalls.h
+++ b/src/lttng-syscalls.h
@@ -42,7 +42,7 @@ struct itimerval;
 struct itimerspec;
 
 #if (LTTNG_LINUX_VERSION_CODE >= LTTNG_KERNEL_VERSION(5,6,0))
-typedef __kernel_old_time_t time_t;
+//typedef __kernel_old_time_t time_t;
 #endif
 
 struct trace_syscall_entry {
diff --git a/src/wrapper/kallsyms.c b/src/wrapper/kallsyms.c
index 6293e07..0e80484 100644
--- a/src/wrapper/kallsyms.c
+++ b/src/wrapper/kallsyms.c
@@ -120,6 +120,7 @@ EXPORT_SYMBOL_GPL(wrapper_kallsyms_lookup_name);
 
 #endif
 
+MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
 #if (LTTNG_LINUX_VERSION_CODE >= LTTNG_KERNEL_VERSION(5,4,0) && defined(CONFIG_ANDROID))
 MODULE_IMPORT_NS(VFS_internal_I_am_really_a_filesystem_and_am_NOT_a_driver);
 #endif
-- 
2.25.1

